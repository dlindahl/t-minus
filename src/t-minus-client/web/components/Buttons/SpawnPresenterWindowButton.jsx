import AppNavButton from './AppNavButton';
import { bindActionCreators } from 'redux';
import { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import OpenInNew from 'react-icons/lib/md/open-in-new';
import { spawnPresenterWindow } from '../../../shared/actions/PresenterActions';

function getState(state) {
  return {
    disabled: !!state.meta.presenterWin
  };
}

function getActions(dispatch) {
  return bindActionCreators({ spawnPresenterWindow }, dispatch);
}

@connect(getState, getActions)
export default class SpawnPresenterWindowButton extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    if(!this.props.disabled) {
      this.props.spawnPresenterWindow();
    }
  }
  render() {
    return (
      <AppNavButton disabled={this.props.disabled} onClick={this.handleClick}>
        <OpenInNew/>
      </AppNavButton>
    );
  }
}

SpawnPresenterWindowButton.propTypes = {
  disabled: PropTypes.bool,
  spawnPresenterWindow: PropTypes.func
};
SpawnPresenterWindowButton.defaultProps = {
  disabled: false
};
