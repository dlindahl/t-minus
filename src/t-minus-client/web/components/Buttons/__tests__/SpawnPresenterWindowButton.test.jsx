import configureStore from 'redux-mock-store';
import emptyObj from 'empty/object';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import SpawnPresenterWindowButton from '../SpawnPresenterWindowButton';

const STATES = {
  default: {
    meta: emptyObj
  }
};
const store = configureStore();

describe('SpawnPresenterWindowButton', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <SpawnPresenterWindowButton store={store(STATES.default)}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('spawns the presenter window when clicked', () => {
    const spawnPresenterWindow = jest.fn();
    const component = renderer.create(
      <Provider store={store()}>
        <SpawnPresenterWindowButton.WrappedComponent
          spawnPresenterWindow={spawnPresenterWindow}
        />
      </Provider>
    );
    const tree = component.toJSON();
    tree.props.onClick();
    expect(spawnPresenterWindow).toHaveBeenCalled();
  });

  it('does not spawn the presenter window when disabled and clicked', () => {
    const spawnPresenterWindow = jest.fn();
    const component = renderer.create(
      <Provider store={store()}>
        <SpawnPresenterWindowButton.WrappedComponent
          disabled
          spawnPresenterWindow={spawnPresenterWindow}
        />
      </Provider>
    );
    const tree = component.toJSON();
    tree.props.onClick();
    expect(spawnPresenterWindow).not.toHaveBeenCalled();
  });
});
