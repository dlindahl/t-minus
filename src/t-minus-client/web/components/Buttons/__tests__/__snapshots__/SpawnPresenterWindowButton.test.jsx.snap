exports[`SpawnPresenterWindowButton renders the component 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "#bdc3c7",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "padding": 0,
      "width": 50,
    }
  }>
  <svg
    fill="currentColor"
    fit={true}
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m23.4 5h11.6v11.6h-3.4v-5.9l-16.3 16.3-2.3-2.3 16.3-16.3h-5.9v-3.4z m8.2 26.6v-11.6h3.4v11.6q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h11.6v3.4h-11.6v23.2h23.2z" />
    </g>
  </svg>
</button>
`;
